google-cloud-bigquery/src/test/java/com/google/cloud/bigquery/it/ITBigQueryTest.java

```diff
@@ -6314,19 +6314,27 @@ public void testStatelessQueries() throws InterruptedException {
 
     // simulate setting the QUERY_PREVIEW_ENABLED environment variable
     bigQuery.getOptions().setQueryPreviewEnabled("TRUE");
-    assertNull(executeSimpleQuery(bigQuery).getJobId());
+    TableResult tableResult = executeSimpleQuery(bigQuery);
+    assertNotNull(tableResult.getQueryId());
+    assertNull(tableResult.getJobId());
 
     // the flag should be case-insensitive
     bigQuery.getOptions().setQueryPreviewEnabled("tRuE");
-    assertNull(executeSimpleQuery(bigQuery).getJobId());
+    tableResult = executeSimpleQuery(bigQuery);
+    assertNotNull(tableResult.getQueryId());
+    assertNull(tableResult.getJobId());
 
     // any other values won't enable optional job creation mode
     bigQuery.getOptions().setQueryPreviewEnabled("test_value");
-    assertNotNull(executeSimpleQuery(bigQuery).getJobId());
+    tableResult = executeSimpleQuery(bigQuery);
+    assertNotNull(tableResult.getQueryId());
```

Comment:
How does this scenario work? Will the jobID be populated when the queryID is populated?
Isn't the jobID and queryID fields an either..or case?

Reply:
It seems like queryId is always populated. As far as I can tell, the queryId == jobId string when jobId is not null.

Reply:
expectations:  
* all queries run through jobs.query statelessly should return a queryId.
* if we run through jobs.query and end up with a stateful (job-based) execution then I currently expect both to be populated.
* If we run via jobs.insert I only expect jobID because queryId isn't part of the job message (or its submessages like statistics).

Reply:
I added an additional test: testTableResultJobIdAndQueryId which runs through the scenario @shollyman described. There is a slight difference in scenario 2 where job.query end up with a stateful execution, JobId is populated and QueryId is not. I think this scenario is consistent with scenario 3 which makes sense.

It seems like the reason we are seeing jobId != null and queryId != null is in the case where a query can be stateless but setQueryPreviewEnabled is false which is covered in testStatelessQueries.

Please take a look.